core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala

```diff
@@ -268,6 +268,8 @@ private object WorkflowFileReader {
   val rawModel = "rawModel"
   val zipModel = "Model.zip"
   def modelStagingDir: String = s"modelStagingDir/model-${System.currentTimeMillis}"
+  val confWithDefaultCodec = new Configuration(false)
+  val codecFactory = new CompressionCodecFactory(confWithDefaultCodec)
```

Comment:
Is it reusable? Is this thread-safe?

Reply:
model reading should be in the driver

Reply:
I think it is. 

The codec factory maps are fully populated after the constructor and val is final. 
Each usage of the WorkflowReader singleton in a different thread necessarily is synchronized with the classloader static class initialization. 

So there is proper JMM happens-before order between the consistent codec factory and the future getCodec lookups.